cmake/modules/kconfig.cmake

```diff
@@ -17,6 +17,8 @@ set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${AUTOCONF_H})
 file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/kconfig/include/generated)
 file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/kconfig/include/config)
 
+set_ifndef(KCONFIG_NAMESPACE "CONFIG")
```

Comment:
Can you make an argument for why we want to reuse this file as-is in sysbuild, with just this change of variables? How much of this is useful general purpose cmake infrastructure for kconfig, in other words?

I ask because this file has 28 occurrences of the string "zephyr". Most of them are just ZEPHYR_BASE, which is fine for sysbuild to reference too, because it is also in the zephyr repository. However, I also see infrastructure for dealing with zephyr modules and other things that don't seem to be generic enough to be useful to sysbuilld.

My concern here is whether using this directly in sysbuild without a bit more refactoring into the "generic" vs. "zephyr-specific" portions of what this file is supposed to do will create maintainability problems.

Reply:
> Can you make an argument for why we want to reuse this file as-is in sysbuild, with just this change of variables? How much of this is useful general purpose cmake infrastructure for kconfig, in other words?

Sure.

We have a couple of things to consider here.
1. Reusability of Zephyr CMake Kconfig module
1. Long term evolution of Zephyr CMake modules (including Kconfig CMake module)
1. Required changes needed for SYSBuild

Now, part of 1. was achieved with #41301 as we can now reuse this is SYSBuild.
The parts we directly want to reuse are:
- Kconfiglib parsing
- Kconfig frontends, like menuconfig, guiconfig, and any custom frontends provided by downstream users
- Long term, we also want the ability for downstream users to also provide SYSBuild Kconfig's in the Zephyr modules

What we don't need at the moment is the `autoconf.h` generation and the current Zephyr module specic Kconfig files.
But for those two parts, the `autoconf.h` does no harm as we are not using it, and the `ZEPHYR_<MODULE_NAME>_KCONFIG` is a variable that is purely exported to kconfiglib, but not used in the SYSBuild Kconfig tree.

Because we do want to support some kind of Zephyr module SYSBuild support later, I decided not to refactor this part, just yet.

Secondly, as 2. above, the Zephyr CMake modules still has enhancements ahead.
First step in #41301 only took files into modules, but their is still improvements to do.
Different CMake modules behaves differently, several modules, for example like `ExternalProject` creates functions that can be used inside the project, other modules, for example `CTest` execute code when included.

What lacks in current Zephyr CMake modules is cleanup inside each module, and to make a clear decision on which approach each module should use.

I prefer to keep this work independent of this PR as it should cover all Zephyr CMake modules, not just kconfig part.

Reply:
OK, that sounds reasonable. Thanks for explaining.